core/shared/src/main/scala/zio/logging/internal/FileWriter.scala

```diff
@@ -15,30 +15,75 @@
  */
 package zio.logging.internal
 
-import java.io.{ BufferedWriter, FileOutputStream, OutputStreamWriter, Writer }
+import zio.logging.FileLoggerConfig
+import zio.logging.FileLoggerConfig.FileRollingPolicy
+
+import java.io.{ BufferedWriter, File, FileOutputStream, OutputStreamWriter, Writer }
 import java.nio.charset.Charset
-import java.nio.file.Path
+import java.nio.file.{ FileSystems, Path }
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 
 private[logging] class FileWriter(
   destination: Path,
   charset: Charset,
   autoFlushBatchSize: Int,
-  bufferedIOSize: Option[Int]
+  bufferedIOSize: Option[Int],
+  rolling: Option[FileLoggerConfig.FileRollingPolicy]
 ) extends Writer {
-  private val writer: Writer = {
-    val output = new OutputStreamWriter(new FileOutputStream(destination.toFile, true), charset)
+  private var currentDestination             = destination
+  private def makeWriter(path: Path): Writer = {
+    val output = new OutputStreamWriter(new FileOutputStream(path.toFile, true), charset)
     bufferedIOSize match {
       case Some(bufferSize) => new BufferedWriter(output, bufferSize)
       case None             => output
     }
   }
+  private var writer: Writer                 = rolling match {
```

Comment:
in case of rollingPolicy, i think some concurrent constructs will have to be used as at same time 2 threads/fibers may try to log, and we do not want to create new writer twice ...

Reply:
I rewrote. Could you check this commit? https://github.com/zio/zio-logging/pull/691/commits/15d17687e4a2edcbc63c678d75e225333899b433